compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Packaging/Metadata.hs

```diff
@@ -40,6 +38,9 @@ data PackageDbMetadata = PackageDbMetadata
   -- We do not bother differentiating between exposed and unexposed modules
   -- since we already warn on non-exposed modules anyway and this
   -- is intended for data-dependencies where everything is exposed.
+  , hashDependencies :: Maybe Int
```

Comment:
Why is this a `Maybe`? Afaict, we always set it to a `Just`.

Reply:
I made it a `Maybe` because otherwise, you'll get an ugly JSON-parsing error if you run the newest SDK version in a project with an already existing package database and metadata file.

Reply:
But if we catch the exception in `readMetadata`, we can also drop the `Maybe` and reinitialize and rewrite the metadata when it fails reading the existing metadata.

Reply:
This actually brings up a good point. We need to include the SDK version and the daml-lf version in the hash as well. Catching the exceptions and reinitializing sounds good.